ksqldb-common/src/main/java/io/confluent/ksql/parser/TokenLocation.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.parser;
+
+import com.google.errorprone.annotations.Immutable;
+import java.util.Objects;
+import java.util.OptionalInt;
+
+/**
+ * Wrapper class for encapsulating the location information of the
+ * antlr Token {@link org.antlr.v4.runtime.Token}
+ */
+@Immutable
+public final class TokenLocation {
+
+  /**
+   * line number returned by {@link org.antlr.v4.runtime.Token#getLine()}
+   */
+  private final OptionalInt line;
```

Comment:
I feel like it's weird to have all the fields here be optional. when is the situation that you would these only _some_ of these fields would be empty? If it's either all the fields or none of them, (e.g. using `TokenLocation.empty()`) then we should just have the field be `Optional<TokenLocation>` in whatever class uses this class.

Also since these are all returned from methods on `Token` why can't we just use the Token class?

Reply:
accepted the first part.

`Token` is not available as a dependency